### PR TITLE
Script for pushing pre-built testing docker images to dockerhub.

### PR DESCRIPTION
**Pull request by jtattermusch**
I've been experimenting a little bit and pushing the prebuilt docker images to docker hub seems fairly easy.
https://hub.docker.com/r/grpctesting/fuzzer_73b2a43e7714a0004a131deb5ef027986976537f/

CC @nicolasnoble

### CODE REVIEW

File: tools/dockerfile/push_testing_images.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+set -ex
+
+cd $(dirname $0)/../..
+git_root=$(pwd)
+cd -
+
+DOCKERHUB_ORGANIZATION=grpctesting
+
+for DOCKERFILE_DIR in tools/dockerfile/test/fuzzer
```
Comment (matt-kwong) [inline tools/dockerfile/push_testing_images.sh]: I think it would be better to not hardcode `tools/dockerfile/test/fuzzer` and make it a .sh argument

**Comment by matt-kwong** [comment]
Docker Hub also supports commit hooks for GitHub, so that images will be updated whenever there is any Dockerfile changes committed to a repo - https://docs.docker.com/engine/tutorials/dockerrepos/#/automated-builds. This might be worth looking into.

File: tools/dockerfile/push_testing_images.sh (hunk 1 above)
Comment (jtattermusch) [inline tools/dockerfile/push_testing_images.sh]: I think eventually there's gonna be a list of N  docker images that need to be uploaded to dockerhub (and one would need to remember which ones). a few other docker images might be added to the list soonish, so I think the current approach is better. We can change it later if needed (I'm sure it's gonna evolve anyway).

**Comment by jtattermusch** [comment]
I knew dockerhub could build Docker images for you, but I though (incorrectly) that your repo structure needs to be setup according to that. We can experiment with autobuilding later, but for now there's a few issues:
-- we need the _SHA suffix for the docker images because we need to make sure we are building against the right version of the dockerfile (pull requests can have modified versions of it)
-- we need to make sure that the image is available before a change is merged into master (because running tests depends on it). So for now, whoever changes the dockerfile (doesn't happen often), is responsible for uploading an updated version of the image to dockerhub.
Again, we can improve the process later on.

**Pull request merged**